Cython/Includes/cpython/datetime.pxd

```diff
@@ -1,27 +1,99 @@
 from cpython.object cimport PyObject
+from cpython.version cimport PY_VERSION_HEX
 
 cdef extern from "Python.h":
     ctypedef struct PyTypeObject:
         pass
 
 cdef extern from "datetime.h":
     """
-    #if PY_MAJOR_VERSION < 3 && !defined(PyDateTime_DELTA_GET_DAYS)
-    #define PyDateTime_DELTA_GET_DAYS(o)         (((PyDateTime_Delta*)o)->days)
+    /* Backport for Python < 3.3 */
+    #ifndef PyDateTime_DELTA_GET_DAYS
+        #define PyDateTime_DELTA_GET_DAYS(o) (((PyDateTime_Delta*)o)->days)
     #endif
-    #if PY_MAJOR_VERSION < 3 && !defined(PyDateTime_DELTA_GET_SECONDS)
-    #define PyDateTime_DELTA_GET_SECONDS(o)      (((PyDateTime_Delta*)o)->seconds)
+    #ifndef PyDateTime_DELTA_GET_SECONDS
+        #define PyDateTime_DELTA_GET_SECONDS(o) (((PyDateTime_Delta*)o)->seconds)
     #endif
-    #if PY_MAJOR_VERSION < 3 && !defined(PyDateTime_DELTA_GET_MICROSECONDS)
-    #define PyDateTime_DELTA_GET_MICROSECONDS(o)   (((PyDateTime_Delta*)o)->microseconds)
+    #ifndef PyDateTime_DELTA_GET_MICROSECONDS
+        #define PyDateTime_DELTA_GET_MICROSECONDS(o) (((PyDateTime_Delta*)o)->microseconds)
+    #endif
+
+    /* Backport for Python < 3.6 */
+    #ifndef PyDateTime_TIME_GET_FOLD
+        #define PyDateTime_TIME_GET_FOLD(o) 0
+    #endif
+    #ifndef PyDateTime_DATE_GET_FOLD
+        #define PyDateTime_DATE_GET_FOLD(o) 0
+    #endif
+
+    /* Backport for Python < 3.6 */
+    #ifndef PyDateTime_FromDateAndTimeAndFold
+        #define __Pyx_DateTime_DateTimeWithFold(year, month, day, hour, minute, second, microsecond, tz, fold) \
```

Comment:
Shouldn't these define the same name? (Also below.)

Reply:
I wasn't sure if we wanted to overwrite it, could be confusing to call `PyDateTime_FromDateAndTimeAndFold` with `tz` when in the `datetime.h` it is not defined to take it. But I am onboard to use a `#undef` and redefine it.

Reply:
No, that's not what I meant. I meant the two different names:
```
#ifndef PyDateTime_FromDateAndTimeAndFold
        #define __Pyx_DateTime_DateTimeWithFold
```
Both should use the name `__Pyx_DateTime_DateTimeWithFold`.

Reply:
Ah yeah, with the Python version check introduced that should be done, let me amend.